subsys/net/ip/ipv4_autoconf.c

```diff
@@ -0,0 +1,280 @@
+/** @file
+ * @brief IPv4 autoconf related functions
+ */
+
+/*
+ * Copyright (c) 2017 Matthias Boesl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if defined(CONFIG_NET_DEBUG_IPV4_AUTOCONF)
+#define SYS_LOG_DOMAIN "net/ipv4_autoconf"
```

Comment:
I think we could have "net/ipv4ll" here instead, it is shorter which is good and still tells what functionality this line corresponds to.